core/modules/wildcard_match.cc

```diff
@@ -335,7 +335,10 @@ int WildcardMatch::DelEntry(struct WmTuple *tuple, wm_hkey_t *key) {
     tuple->ht.Close();
 
     num_tuples_--;
-    memmove(&tuples_[idx], &tuples_[idx + 1],
+
+    // shift remaing tuples to the left
+    memmove(reinterpret_cast<void*>(&tuples_[idx]),
```

Comment:
Perhaps we can update tuples_ (and fields_ as well) with vectors? It should make the code much simpler (although there might be a small penalty with element dereference on the datapath)